### DISCUSSION TITLE
Doing polls the manual way

### DISCUSSION THREAD

**Discussion by alwinm** [alwinm/cholla#7] [Ideas]
react 👍 if acceptable and 👎 to veto

**Comment by alwinm** [comment]
function_like_this

**Reply by alwinm** [reply]
I campaign against this because I have an opinion against it.

**Comment by alwinm** [comment]
FunctionLikeThis

**Comment by alwinm** [comment]
Function_Like_This

**Reply by helenarichie** [reply]
This is the way that a lot of the functions in Cholla are already named, it's easy to read, and the capitalized words make it stand out, which I think is a useful quality.

**Comment by alwinm** [comment]
functionlikethis

**Reply by alwinm** [reply]
Yes, this is the way I want to clearly label my functions for optimal reading.

**Comment by bcaddy** [comment]
Can I suggest that we figure out all our case suggestions at once and start with a base that we perturb rather than going thing-by-thing? I.e. start with the google naming convention then change what we want. I would add to that that I think a formatting style should be chosen first so we can see how our naming convention goes with it.

So I propose we:

1. Choose a format and enforce it with `clang-format`[^1]
2. Choose a naming convention and enforce it with [`clang-tidy`](https://clang.llvm.org/extra/clang-tidy/checks/readability/identifier-naming.html)
3. Add other `clang-tidy` checks as we can get them to pass

## Cases To Choose From
- camelCase
- PascalCase
- snake_case
- Pascal_Snake
- camel_Snake
- MACRO_CASE
- kebab-case
- Train-Case

## Things To Name
- local variables
- global variables (consider replacing with singleton classes)
- Macros
- public member variables
- protected member variables
- private member variables
- kernels
- functions (naming schemes for __device__, __host__, and __device__ __host__?)
- methods
- classes
- Types (PascalCase)
- namespace
- files (python, .cpp, .h, shell, make)

## Other
- globals should be referred to with `::` operator

[^1]: I would like to merge in my MHD stuff before we run clang-tidy so that I don't have merge conflicts on every singe line. This might be a good idea for all of us. Get our code into dev, format, then start more agile developement

**Reply by bcaddy** [reply]
For implementation: I think someone should do a deep dive on `clang-format` and `clang-tidy` and then bring that info back to the group; I anticipate it taking 1-2 weeks. I'm willing to do it after my MHD stuff is passing but I'm not sure when that will be and it seems like we might want to get this ball rolling soon

**Reply by bcaddy** [reply]
https://zed0.co.uk/clang-format-configurator/
